framework/src/executioners/Transient.C

```diff
@@ -395,10 +395,10 @@ Transient::execute()
   if (!_app.halfTransient())
   {
     TIME_SECTION(_final_timer);
-
+    _problem.execMultiApps(EXEC_FINAL);
```

Comment:
@aeslaughter should we move ``execMultiApps(EXEC_FINAL)`` after ``execute(EXEC_FINAL)``?

I tested this input ``moose/modules/stochastic_tools/test/tests/transfers/sampler_postprocessor/master.i`` by adding ``execute_on = FINAL `` for ``SamplerPostprocessorTransfer`` and ``StochasticResults``. It causes a segmentation fault because the ``StochasticResults`` is not called before  ``SamplerPostprocessorTransfer``.

Reply:
This is opening up a can of worms, because what we actually need is PRE_MULTIAPP and POST_MULTIAPP flags in the same capacity we have for UserObjects and AuxKernels.

Basically, depending on what you are doing you might need one or the other.

I am not sure how we should handle this in the short-term, in the long-term we will resolve all these dependencies automatically, but that is really long-term.

Reply:
In other places like `timestep_end`, we call `execute` before `execMultiApps`, should we do the same here? I guess the problem will be gone if we do that.